.github/workflows/build_centos7.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  workflow_dispatch:
+
+jobs:
+  build_centos:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        container: ["centos:7"]
+
+    container:
+      image: ${{ matrix.container }}
```

Comment:
Huh, I didn't expect it to be that convoluted to use CentOS / Fedora on GitHub Actions! Hopefully it gets simpler in the future.

Could we add a comment just explaining this set up is required since GitHub Actions currently don't _officially_ support any Linux OS other than Ubuntu?

Reply:
Sure, it seems like official CentOS support would be much appreciated by GH Actions users, but it's not on their schedule so far (https://github.com/actions/virtual-environments/issues/1443).